src/main/java/de/hs_mannheim/informatik/ct/controller/PrintOutController.java

```diff
@@ -68,45 +77,35 @@
 
     @GetMapping(value = "/rooms")
     public String getRoomPrintoutList(Model model) {
-        model.addAttribute("buildings", buildingService.getAllBuildings());
         return "rooms/roomPrintout";
     }
 
-    @GetMapping(value = "/rooms/{building}")
-    public DeferredResult<ResponseEntity<byte[]>> getRoomPrintout(
-            @PathVariable(value = "building") String building,
-            HttpServletRequest request) {
-        val roomsInBuilding = buildingService.getAllRoomsInBuilding(building);
-        if (roomsInBuilding.size() == 0) {
-            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Building not found or empty");
-        }
 
-        val outFileName = String.format("Gebäude %s.docx", building);
+    //    TODO IMPORTANT this side can be access without being logged in
+    @GetMapping(value = "/rooms/download", produces = "application/zip")
+    public void getRoomPrintout(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+        //setting headers
+        response.setStatus(HttpServletResponse.SC_OK);
+        response.addHeader("Content-Disposition", "attachment; filename=\"allRoomNotes.zip\"");
 
-        val result = new DeferredResult<ResponseEntity<byte[]>>(120 * 1000L);
-        CompletableFuture.runAsync(() -> {
-            try (val buffer = new ByteArrayOutputStream()) {
+        ZipOutputStream zos = new ZipOutputStream(response.getOutputStream());
+
+        for (String building : buildingService.getAllBuildings()) {
```

Comment:
Since every room has its building name as a property, I don't think this is necessary. You can add a `getAllRooms` method to the `RoomService`, the RoomsRepo should have a method for that already. This should make the code cleaner, especially in the `DynamicContentService` class